docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,103 @@
+= {product} Portal API
+
+== Problem
+
+For the xref:explanation/system/details-ui.adoc[{product} Portal] an API endpoint is needed which implements and exposes the backend functionality.
+This API endpoint is described in xref:explanation/system/details-external-services.adoc[External Services] in more detail.
+As documented there, this API endpoint connects various third-party backend systems together and presents them in a well-defined uniform way for the Portal UI or other API clients to be consumed.
+
+It must be possible to connect to various different backend systems which implement the same functionality so that it is a pluggable system and can be re-used by other parties, which might have different backends for the same task.
+
+{product} API only specifies the interface, the implementation is then specific to the provider of the API.
+
+
+== Relevant requirements
+
+* xref:references/quality-requirements/maintainability/portal-api-modularity.adoc[Portal API Modularity]
+* xref:references/quality-requirements/reliability/portal-api-data-recoverability.adoc[Portal API Data Recoverability]
+* xref:references/quality-requirements/performance/portal-api-data-freshness.adoc[Portal API Data Freshness]
+
+== Proposals
+
+=== Custom API
+
+For each subsystem there will be well-defined interfaces that serve as "contracts" which providers have to implement.
+The API (`api.appuio.cloud`) is a custom (REST or GraphQL) API endpoint which consumes third-party providers.
+The providers have to behave according to the contract, but may need to connect (or "adapt") to their concrete subsystem via API.
+Authentication, authorization and other core API features will have to be implemented specific within this API.
+Operations can be a synchronous or asynchronous.
+
+With a well-defined interface, clients can be generated for all major programming languages, for example for frontends or CLI apps.
+
+image::decision/rest-api-proposal.drawio.svg[]
+
+.Invoicing example
+[example]
+====
+. At the end of a month, the Portal API collects metrics to prepare a new invoice.
```

Comment:
That's not what the API is used for. This task is part of the to-be-defined metering/billing system.
Can we find a better example?

Reply:
I understood the `api.appuio.cloud` to be the business logic or orchestration layer for all things, not just the frontend...

Reply:
Nope, that's not the idea of the API. The API is the interconnecting point between external systems. The whole architecture is meant to be implemented in microservices.

Reply:
I'm sorry, here we have entirely different understandings of "API". I don't see any requirement saying or justifying "it has to be microservices". I don't see a need to "enforce" the complexity of microservices architecture unless there are good reasons for it.
I think we need to define new terms (not using "API" please) in the glossary and agree on the same language first before discussing this further.